modin/numpy/arr.py

```diff
@@ -1020,74 +1106,94 @@ def prod(
         if self._ndim == 1:
             if axis == 1:
                 raise numpy.AxisError(1, 1)
-            result = self._query_compiler.astype(
-                {col_name: out_dtype for col_name in self._query_compiler.columns}
+            result = self._query_compiler
+            if isinstance(where, array):
+                result = where.where(self, 1)._query_compiler
+            result = result.astype(
```

Comment:
I will stop my suggestion of `target = where.where(...) if foo else self` from now on, but it does apply here and in other places